README.md

```diff
@@ -1,4 +1,241 @@
+
+It is the responsibility of our program as a whole to create and establish the database. Accordingly, you'll see our Ruby programs set up such that they have a `config` directory that contains an `environment.rb` file. This file will look something like this:
+
+```ruby
+require 'sqlite3'
+require_relative '../lib/song.rb'
+
+DB = {:conn => SQLite3::Database.new("db/music.db")}
+```
+
+Here we set up a constant, `DB`, that is equal to a hash that contains our connection to the database. In our `lib/song.rb` file, we can therefore access the `DB` constant and the database connection it holds like this:
+
+```ruby
+DB[:conn]
+```
+
+So, as we move through this reading, let's assume that our hypothetical program has just such a `config/environment.rb` file and that the `DB[:conn]` constant refers to our connection to the database. 
+
+Now that our hypothetical database is set up in our hypothetical program, let's move on to our `Song` class and it's equivalent database table. 
+
+### Creating the Table
+
+According to the ORM convention in which a class is mapped to or equated with a database table, we need to create a songs table. We will accomplish this by writing a class method in our `Song` class that creates this table. 
+
+**To "map" our class to a database table, we will create a table with the same name as our class and give that table column names that match the `attr_accessor`s of our class.**
+
+Here's an example of a `Song` class that maps instance attributes to table columns:
+
+```ruby
+class Song
+
+  attr_accessor :name, :album
+  
+  attr_reader :id
+  
+  def initialize(name, album, id=nil)
+    @id = id
+    @name = name
+    @album = album
+  end
+  
+  def self.create_table
+    sql =  <<- SQL 
+      CREATE TABLE IF NOT EXISTS songs (
+        id PRIMARY KEY INTEGER, 
+        name TEXT, 
+        album TEXT
+        )
+        SQL
+    DB[:conn].execute(sql) 
+  end
+
+end
+```
+
+Let's break down this code.
+
+#### The `id` Attribute
+
+Notice that we are initializing an individual `Song` instance with an `id` attribute that has a default value of `nil`. Why are we doing this? First of all, songs need an `id` attribute only because they will saved into the database and we know that each table row needs an `id` value which is the primary key. 
+
+When we create a new song with the `Song.new` method, we *do not set that song's id*. A song gets an `id` only when it gets saved into the database (more on inserting songs into the database later). We therefore set the default value of the `id` argument that the `#initialize` method takes equal to `nil`, so that we can create new song instances that *do not have an `id` value. We'll leave that up to the database to handle later on. 
```

Comment:
Add something reminding them why the DB needs to set the id.the id needs to be unique. Only the database is able to ensure that